test_crates/exhaustive_struct_added/new/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+#![no_std]
+
+// Existing struct should not trigger the lint
+pub struct ExistingStruct {
+    pub bar: u64,
+}
+
+// New non-exhaustive struct with public fields should trigger the lint
+pub struct NewStruct {
+    pub bar : u64,
+}
+
+// This struct is not public, so it should not trigger the lint.
+struct PrivateStruct {
+    foo: u64,
+}
+
+// This struct have non-public fields , so it should not trigger the lint.
+pub struct StructWithPrivateField {
+    foo : u64
+}
+
+// This struct is #[non_exhaustive], so it should not trigger this lint
+#[non_exhaustive]
+pub struct NonExhaustiveStruct {
+    bar: u64,
+}
```

Comment:
```suggestion
}

// This struct shouldn't trigger the lint because it cannot be constructed
// with a literal within the public API of the crate.
pub struct StructWithHiddenField {
    pub a: i64,

    #[doc(hidden)]
    pub b: i64,
}
```